cli/src/commands/commit.rs

```diff
@@ -50,6 +51,16 @@ pub(crate) struct CommitArgs {
     /// $ JJ_USER='Foo Bar' JJ_EMAIL=foo@bar.com jj commit --reset-author
     #[arg(long)]
     reset_author: bool,
+    /// Set author to the provided string
+    ///
+    /// This changes author name and email while retaining author
+    /// timestamp if the description is filled.
```

Comment:
nit: "if the description is filled" isn't accurate. Maybe we can say "discardable" or "empty commit with no description" (copied from 1eebbe57c080568eec96e548c02ab8e43d8b1d8a)

(also applies to `cmd_describe()`)

Reply:
I think you meant negation of these statements? Author timestamps of discardable commits have no value.
Changed to `non-discardable commits`.

Reply:
Looks good, thanks.